syntax = "proto3";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

// to compile
// python3 -m grpc_tools.protoc -I./ --python_out=. --grpc_python_out=. --pyi_out=. ./meteoServer.proto

message RawMeteoData {
  string id = 1;
  float temperature = 2;
  float humidity = 3;
  google.protobuf.Timestamp timestamp = 4;
}

message RawPollutionData {
  string id = 1;
  int32 co2 = 2;
  google.protobuf.Timestamp timestamp = 3;
}


service MeteoService {
  rpc SendMeteoData(RawMeteoData) returns (google.protobuf.Empty){}
  rpc SendPollutionData(RawPollutionData) returns (google.protobuf.Empty){}
}